docs/plugins/best_practices.md

```diff
@@ -0,0 +1,175 @@
+# Best practices when developing napari plugins
+
+There are a number of good and bad practices that may not be immediately obvious
+when developing a plugin.  This page
+
+## Do
+
+- **Update your README to make it immediately clear what your plugin does in a
+  few sentences** 
+
+    This is your elevator pitch. You'll loose a lot of potential users quickly
+    if they have to look in the source code, or look through a large
+    documentation page, just to know what the gist of the plugin is.
+
+- **Use images, gifs, or movies to *show* what your plugin does**. 
+
+  Images and movies are *extremely* useful for conveying the usefulness of your
+  plugin. Both github and napari hub make it easy to embed videos in your
+  readme. For example, if you have a dock widget plugin that plots something as
```

Comment:
```suggestion
readme. Github can be used to host the video by posting a `.gif` file in a comment, 
issue or pull request on your repository before linking to it in the readme. 
For example, if you have a dock widget plugin that plots something as
```

Reply:
I like this pattern and wouldn't have thought of it myself - not sure if we should be recommending it though so maybe not!

Reply:
I've actually just removed that bit from this guide... remember that though, for the "style guide" proposed above!